src/signatureUtils.js

```diff
@@ -0,0 +1,253 @@
+ * @returns {boolean} true if either signatures are valid or false
+ */
+async function verifySignature(digitalSignatures, signedPayload, publicKeys, recipientClientId) {
+    let result, publicKey
+    try {
+        for (i = 0; i < digitalSignatures.length; i++) {
+            publicKey = await createCryptoPublicKey(publicKeys[i])
+            result = await cryptoVerify(digitalSignatures[i], publicKey, signedPayload)
+            if (result) {
+                return result
+            }
+        }
+    } catch (error) {
+        logger.error('error occured while verifying digital signature for client id %s due to %s ', recipientClientId, error.message)
+        return false
+    }
+}
+
+/**
+ * Crypto verifies the signature of the I/O Events signed payload
+ * 
+ * @param {*} signature I/O Events digital signature
+ * @param {*} pubKey I/O Events public key
+ * @param {*} signedPayload I/O Events signed payload 
+ * @returns {boolean} true or false
+ */
+async function cryptoVerify(signature, pubKey, signedPayload) {
+    try {
+        return crypto.verify('rsa-sha256', new TextEncoder().encode(signedPayload), pubKey, Buffer.from(signature, 'base64'))
+    } catch (error) {
+        logger.error('error during crypto verification of digital signature due to => ' + error.message)
+        return false
+    }
+}
+
+async function createCryptoPublicKey(publicKey) {
+    return crypto.createPublicKey(
+        {
+            key: publicKey,
+            format: 'pem',
+            type: 'spki'
+        })
+}
+
+/**
+ * Checks if the recipient client id is the valid target recipient of the event payload
+ * 
+ * @param {*} decodedJsonPayload a valid json event payload used in I/O Events signing
+ * @param {*} recipientClientId target recipient client id
+ * @returns {boolean} true if valid target recipient or false 
+ */
+function isTargetRecipient(decodedJsonPayload, recipientClientId) {
+    const targetRecipient = decodedJsonPayload.recipient_client_id
+    if (targetRecipient !== null && typeof (targetRecipient) !== 'undefined') {
+        return targetRecipient === recipientClientId
+    }
+    logger.error('target recipient client id is either null or missing')
+    return false
+}
+
+/**
+ * Generates generic error json response object based on HTTP error codes and message
+ * 
+ * @param {*} statusCode HTTP error codes
+ * @param {*} message custom error message
+ */
+function genErrorResponse(statusCode, message) {
+    const response = {
+        statusCode: statusCode,
+        body: message,
+        headers: {
+            'Content-Type': 'application/json'
+        }
+    }
+    return {
+        error: response
+    }
+}
+
+module.exports = {
+    verifyDigitalSignature,
+    isTargetRecipient,
+    genErrorResponse,
+    getProperPayload,
+    fetchPemEncodedPublicKeys,
+    cryptoVerify,
+    verifySignature,
+    fetchPublicKeyFromCloudFront,
+    getPubKeyFileName,
+    getKeyFromCache
```

Comment:
Why do we need to export all functions here? We should only be exporting public functions that are used in other files?

Reply:
these are exported to be used in tests `signatureUtilsTest.js`, will remove this one which is not getting used `getPubKeyFileName`

Reply:
Hmm ok. I wonder if that's a good practice, but since this is time sensitive, I think we are ok.